src/cache.js

```diff
@@ -0,0 +1,131 @@
+import window from 'global/window';
+import {getUrl} from './url';
+
+// Tracks previously-downloaded scripts and/or detected players.
+//
+// The keys follow the format "accountId_playerId_embedId" where accountId is
+// optional and defaults to "*". This happens when we detect pre-existing
+// player globals.
+const cache = new window.Map();
+
+/**
+ * Get the cache key given some properties.
+ *
+ * @private
+ * @param  {object} props
+ *         Properties describing the player record to cache.
+ *
+ * @param  {string} props.playerId
+ *         A player ID.
+ *
+ * @param  {string} props.embedId
+ *         An embed ID.
+ *
+ * @param  {string} [props.accountId="*"]
+ *         An optional account ID. This is optional because when we search for
+ *         pre-existing players to avoid downloads, we will not necessarily
+ *         know the account ID.
+ *
+ * @return {string}
+ *         A key to be used in the script cache.
+ */
+const key = ({accountId, playerId, embedId}) => `${accountId || '*'}_${playerId}_${embedId}`;
```

Comment:
Should we just use the object as the key rather than having this function? Seems like maps allow objects as keys instead of just strings.

Reply:
But then we'd have to check deep-equality rather than by reference. Each time we pass an object to `cache.store`, it's a different object.